kyaml/fn/framework/gosdk/kubeobject.go

```diff
@@ -0,0 +1,293 @@
+package gosdk
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"sigs.k8s.io/kustomize/kyaml/fn/framework/gosdk/unstructured"
+	"sigs.k8s.io/kustomize/kyaml/kio/kioutil"
+)
+
+// KubeObject presents a k8s object.
+type KubeObject struct {
```

Comment:
This type has a lot of similar methods as `Resource` in `api/resource/resource.go`, which wraps an `RNode` which wraps a yaml Node.

This seems like the same pattern that you have here - `KubeObject` wraps a `MapVariant` which wraps a yaml Node. 

I'm wondering, would it make sense to move `Resource` into kyaml, and add the methods you've added in this PR to `Resource` and `RNode` instead of defining new wrapper types? Pros/cons?